app/models/orchestration_template.rb

```diff
@@ -20,6 +20,25 @@ class OrchestrationTemplate < ApplicationRecord
   attr_accessor :remote_proxy
   alias remote_proxy? remote_proxy
 
+  DEPRECATED_TYPES = {
+    'OrchestrationTemplateCfn'   => 'ManageIQ::Providers::Amazon::CloudManager::OrchestrationTemplate',
+    'OrchestrationTemplateHot'   => 'ManageIQ::Providers::Openstack::CloudManager::OrchestrationTemplate',
+    'OrchestrationTemplateVnfd'  => 'ManageIQ::Providers::Openstack::CloudManager::VnfdTemplate',
+    'OrchestrationTemplateAzure' => 'ManageIQ::Providers::Azure::CloudManager::OrchestrationTemplate',
+  }.freeze
+
+  # Temporarily maintain backward compatibility for API users. Should be removed later.
```

Comment:
Should this live in the API?

Reply:
I consulted with @abellotti. API controller automatically maps a request to backend class. It does not have an explicit place for class `OrchestrationTemplate` where we can add the conversion.

Reply:
Yes this should live in the API. Why would you put this concern here when it's literally an API interface issue?

> It does not have an explicit place for class OrchestrationTemplate where we can add the conversion.

That's not true. Wherever in the bowels of the BaseController we do collection => class convention, this exact sort of thing can live.

Let the API dictate it's own interface and interpret from the backend what the consumer is asking for to the models the consumer actually wants.

Reply:
@bzwei BTW please use the public Gitter room(s) to consult about such things, as then the REST of the API team can assist with any confusion (including @abellotti! ❤️). We remotees would surely appreciate it.